### ISSUE TITLE
problem sending to sabnzbd when using reverse proxy

### ISSUE THREAD

**Issue by JohanE67** [spotweb/spotweb#844]
I am running spotweb in a Docker container, using the image erikdevries/spotweb. I also have an nginx reverse proxy running and am normally running through this. Everything appears to be working just fine, except for adding nzb's to sabnzbd queue. 
Spotweb and the reverse proxy are running on a remote vps. Sabnzbd is running on my home server. Spotweb is up-to-date with git, sabnzbd is version 4.3.1.

When I access spotweb directly, using http://vps.mydomain.nl:8085, everything works as expected. I filtered to "books", then clicked the download arrow. Here's what I see in the sabnzbd log file:
```
2024-05-04 11:47:08,665::INFO::[notifier:142] Sending notification: Download aan wachtrij toegevoegd - Probeer NZB op te halen van http://vps.mydomain.nl:8085/?page=getnzb&action=display&messageid=1Q2VtWfVUPQBsg1ZgTQHA@spot.net&apikey=myworkingapikey (type=download, job_cat=books)
2024-05-04 11:47:08,669::INFO::[urlgrabber:120] Grabbing URL http://vps.mydomain.nl:8085/?page=getnzb&action=display&messageid=1Q2VtWfVUPQBsg1ZgTQHA@spot.net&apikey=myworkingapikey
```
The download starts and the green check mark appears in spotweb.

However, when I access spotweb through the reverse proxy, using https://spotweb.mydomain.nl, initially everything looks fine. I again filter to "books", then click the download arrow again. Here's what I now see in the sabnzbd log file:

```
2024-05-04 11:34:06,895::INFO::[notifier:142] Sending notification: Download aan wachtrij toegevoegd - Probeer NZB op te halen van https://spotweb.mydomain.nl/?page=index&search[tree]=cat0_z2&search[value][]=NULL/?page=getnzb&action=display&messageid=1Q2VtWfVUPQBsg1ZgTQHA@spot.net&apikey=myworkingapikey (type=download, job_cat=books)
2024-05-04 11:34:06,899::INFO::[urlgrabber:120] Grabbing URL https://spotweb.mydomain.nl/?page=index&search[tree]=cat0_z2&search[value][]=NULL/?page=getnzb&action=display&messageid=1Q2VtWfVUPQBsg1ZgTQHA@spot.net&apikey=myworkingapikey
2024-05-04 11:34:07,255::INFO::[urlgrabber:274] Unknown filetype when fetching NZB, retry after 30s https://spotweb.mydomain.nl/?page=index&search[tree]=cat0_z2&search[value][]=NULL/?page=getnzb&action=display&messageid=1Q2VtWfVUPQBsg1ZgTQHA@spot.net&apikey=myworkingapikey
```

So, in this case, the search string from the URL (`?page=index&search[tree]=cat0_z2&search[value][]=NULL`) gets embedded in the call toe sabnzbd, leading to sabnzbd being unable to download the nzb file. After a couple of retries it gives up. Also, the red cross appears immediately in spotweb.

Anyone can give me a hint on how to solve this?
Thanks, Johan

**Comment by mesa57** [comment]
sabnzbd creates an invalid url ? Have you asked the sabnzbd guys what could be the cause ?

**Comment by Sweepr** [comment]
I guess looking at this the problem lies within the revere proxy configuration, i think there is a URL rewrite that isn't working properly.

Without reverse proxy:

http://vps.mydomain.nl:8085/?page=getnzb&action=display&messageid=1Q2VtWfVUPQBsg1ZgTQHA@spot.net&apikey=myworkingapikey (type=download, job_cat=books)

With reverse proxy:

Fetch NZB: https://spotweb.mydomain.nl**_/?page=index&search[tree]=cat0_z2&search[value][]=NULL_**/?page=getnzb&action=display&messageid=1Q2VtWfVUPQBsg1ZgTQHA@spot.net&apikey=myworkingapikey (type=download, job_cat=books)

The **_/?page=index&search[tree]=cat0_z2&search[value][]=NULL_** is somewhere added to the URL string, this is not coming from Spotweb or SabNZBd

**Comment by JohanE67** [comment]
> 
> The **_/?page=index&search[tree]=cat0_z2&search[value][]=NULL_** is somewhere added to the URL string, this is not coming from Spotweb or SabNZBd

That string is what I get in Spotweb as part of the url when I click the "Books"filter. It does that regardless of whether I go in through the reverse proxy or not. See screen shots below:

![with_proxy](https://github.com/spotweb/spotweb/assets/11681035/bd06c8a2-11bd-465a-b58d-4d820e841d4f)

![Without_proxy](https://github.com/spotweb/spotweb/assets/11681035/b62e0eb4-7e3f-488a-9cf8-dd4465f76d47)

The difference is only that, when I'm behind the proxy, this string gets included in the request that's sent to sabnzbd. Sabnzbd interprets that part of the request as the command to send back to spotweb to retrieve the nzb. From SpotWeb log:

```
172.18.0.3 - - [06/May/2024:08:28:47 +0200] "GET /?page=index&search[tree]=cat0_z2&search[value][]=NULL/?page=getnzb&action=display&messageid=8gudqJAJqVAuG44ZgN1AN@spot.net&apikey=0e97f27405d233629ab0ca62efc77d05 HTTP/1.1" 200 415 "-" "SABnzbd/4.3.1"
172.18.0.3 - - [06/May/2024:08:29:18 +0200] "GET /?page=index&search[tree]=cat0_z2&search[value][]=NULL/?page=getnzb&action=display&messageid=8gudqJAJqVAuG44ZgN1AN@spot.net&apikey=0e97f27405d233629ab0ca62efc77d05 HTTP/1.1" 200 415 "-" "SABnzbd/4.3.1"
172.18.0.3 - - [06/May/2024:08:29:48 +0200] "GET /?page=index&search[tree]=cat0_z2&search[value][]=NULL/?page=getnzb&action=display&messageid=8gudqJAJqVAuG44ZgN1AN@spot.net&apikey=0e97f27405d233629ab0ca62efc77d05 HTTP/1.1" 200 415 "-" "SABnzbd/4.3.1"
```

I cannot find a log file containing the actual command that spotweb sends to sabnzbd. Perhaps someone can point me to where to find this?

When I go in directly, it's not included and only the correct request is sent.

I have no rewrite statement in my proxy setup.

Regards,
Johan

**Comment by mesa57** [comment]
The log shows what sabnzbd sends to spotweb, but not how sabnzbd retrieved the information to do that. Maybe there is a problem on that site ?

**Comment by JohanE67** [comment]
As far as I can tell, Spotweb sends a request to sabnzbd, containing the url where sabnzbd should get the nzb to download.
This url is supposed to be this: `https://spotweb.mydomain.nl/?page=getnzb&action=display&messageid=1Q2VtWfVUPQBsg1ZgTQHA@spot.net&apikey=myworkingapikey`, like the request that's being sent when not using the reverse proxy.

However, the search string that's being used in the spotweb url at the time of sending the request gets somehow included in the request sent to sabnzbd. Unfortunately I can't find a log containing the actual request that spotweb is sending out.

**Comment by mesa57** [comment]
What action spotweb do you perform to send the request to sabnzbd ?

**Comment by JohanE67** [comment]
I click the green down pointing arrow at the far right of the line of the spot I want to download.

**Comment by mesa57** [comment]
I've tested this : spotweb on a vps server secured with https and sabnzbd on a vm at home and configured spotweb to send the nzb to my home ip.
Everything works as expected, the download starts.
Http analyser shows following incoming request : 
`POST /api?mode=addfile&apikey=myapikey&output=json&nzbname=MasterChef+Australia+-+S016E10&cat=tv`

Ergo: nothing wrong with spotweb, somehow you've messed up you're configuration.

**Comment by JohanE67** [comment]
Only difference there is that I use the option "Call SABnzbd via the users' browser". The option "HTTP through SpotWeb" is not working for me. When I choose that and try to start a download, I get a popup message containing "garbage":
![Error](https://github.com/spotweb/spotweb/assets/11681035/9caa1213-280c-40fa-8653-107ac28c01e6)

_Nothing_ is getting sent to sabnzbd in this case.

Still, this all happens only when using the reverse proxy. When using direct entry (no https), both methods work.

**Comment by mesa57** [comment]
Sorry, i cannot reproduce that behaviour. Spent more than 4 hours to check if there are some errors in spotweb, but found none. I think you should get advise from experts on reverse proxy's,

**Comment by JohanE67** [comment]
After a lot of digging and trying, I found that all problems were caused by having this statement in the "location" block of the nginx config: `proxy_set_header X-Forwarded-Uri $request_uri;`. Removed that and everything works as expected.

Why was it there in the first place? I use the [swag](https://docs.linuxserver.io/general/swag/) container to provide the reverse proxy. This comes with prepared proxy configurations for a lot of applications. Each config has a line to "include" a [proxy.conf file](https://github.com/linuxserver/docker-swag/blob/master/root/defaults/nginx/proxy.conf.sample) that holds many "standard" proxy settings. The mentioned one was included there. Didn't cause any trouble on my other containers, but apparently, SpotWeb doesn't like it.
